### PR TITLE
Avoid adding multiple NETWORK_INFO_EVENT events causing memory leak.

### LINKED ISSUE #2505: MaxListenersExeededWarning: Possible EventEmitter memory leak detected NETWORK_INFO_CHANGED
## Description
Every time I disable internet on my mobile device for a few seconds (airplane mode), I got this warning:
`MaxListenersExeededWarning: Possible EventEmitter memory leak detected. 11 NETWORK_INFO_CHANGED listeners added. Use emitter.setMaxListeners() to increase limit`

## Current behavior
IMG_6361
IMG_6362

## Expected Behavior
No max listeners exeeded warning.

## Possible Solution
Remove previous listeners before adding new one.

## Steps to reproduce
Disable internet for 5 sec and enable it again (airplane mode).

# Environment details
react-native@0.73.6

### PR DESCRIPTION
**Pull request by Romick2005**
This fixes #2505.

### CODE REVIEW

File: modules/xmpp/ResumeTask.js [hunk 1]
```diff
@@ -49,7 +49,9 @@ export default class ResumeTask {
         this._cancelResume();
 
         this._resumeRetryN += 1;
-
+        if (this._networkOnlineListener) {
```
Comment (saghul) [inline modules/xmpp/ResumeTask.js:52]: Rather than here, I think the code currently in cancel should move to `_cancelResume` so it applies in both cases.

**Review by saghul — CHANGES_REQUESTED** [review]

File: modules/xmpp/ResumeTask.js (hunk 1 above)
Comment (Romick2005) [reply modules/xmpp/ResumeTask.js]: Agree. Applied MR suggestion.

**Review by saghul — DISMISSED** [review]

**Comment by saghul** [comment]
Hum, the test failures seem to be legit.

**Comment by saghul** [comment]
Jenkins please test this please.

**Comment by Romick2005** [comment]
Is there anything I need to do for this to be merged?

**Review by saghul — APPROVED** [review]

**Pull request merged**